### PR TITLE
[SM-568] Delete service accounts

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Implement delete service accounts. Uses the same logic as folders.

## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/web/src/locales/en/messages.json [hunk 1]
```diff
@@ -5815,6 +5815,33 @@
   "viewServiceAccount":{
     "message":"View service account"
   },
+  "deleteServiceAccountDialogMessage": {
+    "message":  "Deleting service account $PROJECT$ is permanent and irreversible.",
+    "placeholders": {
+      "project": {
+        "content": "$1",
+        "example": "project name"
+      }
```
Comment (willmartian) [inline apps/web/src/locales/en/messages.json:5824]: Placeholder should reference service account, not project.

**Review by willmartian — DISMISSED** [review]
Looks good 🥳

 A couple of small items:

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/dialog/service-account-delete-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  }
+
+  get dialogContent() {
+    return this.data.serviceAccounts.length === 1
+      ? this.i18nService.t("deleteServiceAccountDialogMessage", this.data.serviceAccounts[0].name)
+      : this.i18nService.t("deleteServiceAccountsDialogMessage");
+  }
+
+  get dialogConfirmationLabel() {
+    return this.i18nService.t("deleteProjectInputLabel", this.dialogConfirmationMessage);
```
[the remark this answers is not in the document]
Comment (Hinton) [reply bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/dialog/service-account-delete-dialog.component.ts]: I agree.

File: apps/web/src/locales/en/messages.json [hunk 3]
```diff
@@ -5845,6 +5845,33 @@
     "message": "View service account",
     "description": "Action to view the details of a service account."
   },
+  "deleteServiceAccountDialogMessage": {
+    "message":  "Deleting service account $PROJECT$ is permanent and irreversible.",
```
Comment (Thomas-Avery) [inline apps/web/src/locales/en/messages.json:5849]: `Deleting service account $PROJECT$`

`Deleting service account $SERVICEACCOUNT$`

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts [hunk 4]
```diff
@@ -54,6 +55,25 @@ export class ServiceAccountService {
...
+  async delete(projects: ServiceAccountView[]): Promise<BulkOperationStatus[]> {
+    const projectIds = projects.map((project) => project.id);
+    const r = await this.apiService.send(
+      "POST",
+      "/service-accounts/delete",
+      projectIds,
+      true,
+      true
+    );
+    this._serviceAccount.next(null);
```
Comment (Thomas-Avery) [inline bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts:67]: `projects: ServiceAccountView[]`

Probably want to rename this to `serviceAccounts`. Reads like we are deleting projects.

File: apps/web/src/locales/en/messages.json [hunk 5]
```diff
@@ -5845,6 +5845,33 @@
...
+    "placeholders": {
+      "project": {
```
Comment (Thomas-Avery) [inline apps/web/src/locales/en/messages.json:5851]: `project` -> `serviceAccount`

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts [hunk 6]
```diff
@@ -54,6 +55,25 @@ export class ServiceAccountService {
...
+    return r.data.map((element: { id: string; error: string }) => {
+      const bulkOperationStatus = new BulkOperationStatus();
+      bulkOperationStatus.id = element.id;
+      bulkOperationStatus.name = projects.find((project) => project.id == element.id).name;
```
Comment (Thomas-Avery) [inline bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts:71]: `project` -> `serviceAccount`

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]
Looks good, just a few minor symbol changes for readability.

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts [hunk 7]
```diff
@@ -54,6 +55,25 @@ export class ServiceAccountService {
...
 
+  async delete(serviceAccounts: ServiceAccountView[]): Promise<BulkOperationStatus[]> {
+    const projectIds = serviceAccounts.map((project) => project.id);
+    const r = await this.apiService.send(
+      "POST",
+      "/service-accounts/delete",
+      projectIds,
+      true,
+      true
+    );
```
Comment (Thomas-Avery) [inline bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts:66]:
```suggestion
    const serviceAccountIds = serviceAccounts.map((serviceAccount) => serviceAccount.id);
    const r = await this.apiService.send(
      "POST",
      "/service-accounts/delete",
      serviceAccountIds,
      true,
      true
    );
```

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts [hunk 8]
```diff
@@ -54,6 +55,25 @@ export class ServiceAccountService {
...
+    this._serviceAccount.next(null);
+    return r.data.map((element: { id: string; error: string }) => {
+      const bulkOperationStatus = new BulkOperationStatus();
+      bulkOperationStatus.id = element.id;
+      bulkOperationStatus.name = serviceAccounts.find((project) => project.id == element.id).name;
```
Comment (Thomas-Avery) [inline bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts:71]:
```suggestion
      bulkOperationStatus.name = serviceAccounts.find((serviceAccount) => serviceAccount.id == element.id).name;
```

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts (hunk 7 above)
Comment (Hinton) [reply bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.service.ts]: Sorry, I mixed it up and thought we for some reason were fetching project ids for a service account.

**Review by Thomas-Avery — APPROVED** [review]
LGTM

**Pull request merged**